extensions/typescript-language-features/src/languageFeatures/copilotRelated.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { nulToken } from '../utils/cancellation';
+import { jsTsLanguageModes, isSupportedLanguageMode } from '../configuration/languageIds';
+import { DocumentSelector } from '../configuration/documentSelector';
+import { API } from '../tsServer/api';
+import type * as Proto from '../tsServer/protocol/protocol';
+import { ITypeScriptServiceClient } from '../typescriptService';
+import { conditionalRegistration, requireMinVersion } from './util/dependentRegistration';
+
+const minVersion = API.v570;
+const dummyDisposable = new vscode.Disposable(() => { });
+
+export function register(
+	_selector: DocumentSelector,
+	client: ITypeScriptServiceClient,
+) {
+	return conditionalRegistration([
+		requireMinVersion(client, minVersion),
+	], () => {
+		const ext = vscode.extensions.getExtension('github.copilot');
+		if (!ext) {
+			return dummyDisposable;
+		}
+		ext.activate().then(() => {
+			const relatedAPI = ext.exports as {
+				registerRelatedFilesProvider(
+					providerId: { extensionId: string; languageId: string },
+					callback: (uri: vscode.Uri) => Promise<{ entries: vscode.Uri[]; traits?: { name: string; value: string }[] }>
+				): void;
+			} | undefined;
+			if (relatedAPI?.registerRelatedFilesProvider) {
+				for (const languageId of jsTsLanguageModes) {
```

Comment:
Can we use the `selector` passed in here instead? You can also update `register` to take a list of language ids

Reply:
I checked the code path leading to `selector` and it includes the languages in the `syntax` property.